### ISSUE TITLE
Automatically redirect /amp/ endpoint to ?amp on 404 responses

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#4466]
## Feature description

When using another plugin that tries to use `/amp/` for all URLs, switching to this AMP plugin can result in non-singular URLs 404ing. To deal with that, I created a helper migration plugin: https://gist.github.com/westonruter/6b7d3b168711ee03c4ee29b68c3719b2

Nevertheless, this seems like something that would make sense to be part of the the AMP plugin directly. In particular, it would make sense to add after #2204 is figured out.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
A related issue came up in https://github.com/ampproject/amp-wp/issues/474#issuecomment-609764881.

Namely, if someone originally had `/amp/` for pages, upon switching to the AMP plugin these URLs won't work anymore and they may end up getting routed to WordPress's own canonical redirect system, so in that case `/some-page/amp/` will redirect to `/some-page/` and not `/some-page/?amp` as expected. While we don't want to register an `amp` endpoint for pages, we should perhaps hook into WordPress's canonical redirects to check for `/amp/` being present, and make sure that WordPress will redirect to `?amp` (or `?amp=1` per #2204) instead of removing it from the URL altogether.

**Comment by westonruter** [comment]
Closing as duplicate of #2062
